src/Http/Headers/src/ContentDispositionHeaderValue.cs

```diff
@@ -625,48 +626,55 @@ private static string Encode5987(StringSegment input)
             ? stackalloc byte[MaxStackAllocSizeBytes]
             : bufferFromPool = ArrayPool<byte>.Shared.Rent(maxInputBytes);
 
+        var maxInputAsciiCount = Encoding.ASCII.GetMaxCharCount(input.Length);
+        char[]? charBufferFromPool = null;
+        Span<char> tempCharBuffer = maxInputAsciiCount <= MaxStackAllocSizeChars
+            ? stackalloc char[MaxStackAllocSizeChars]
```

Comment:
I'm still curious why this wouldn't be `maxInputAsciiCount`, as opposed to `MaxStackAllocSizeChars`.  Isn't the former smaller?

Reply:
Yes, of course. The copy-paste issue from above.

Reply:
It seems on my measurements that using a compile time MaxStackAllocSizeChars has a smaller perf benefit.

Reply:
This code segment is removed.